### PR TITLE
badges-cy-109

### PR DESCRIPTION
**Pull request by steiningermona**
I had already opened a PR but I noticed that the resizing of the badge had to be done differently (in comparison to before) if the badge contains more than one character

### CODE REVIEW

File: src/components/badge/Badge.tsx [hunk 1]
```diff
@@ -1,40 +1,27 @@
 import React from "react";
+import { EBadgePosition } from "./EBadgePosition";
 import { SBadge } from "./styles/SBadge";
-import { SBadgeIconContent } from "./styles/SBadgeIconContent";
 import { SBadgeIcon } from "./styles/SBadgeIcon";
-import { EBadgePosition } from "./EBadgePosition";
+import { SBadgeIconContent } from "./styles/SBadgeIconContent";
+
+export type IBadgeType = "regular" | "important" | "white" | "dark" | "color";
+export type IBadgeSizeType = "micro" | "small" | "medium";
```
Comment (diegomart2000) [inline src/components/badge/Badge.tsx]: @steiningermona  Coordinate with @nicolasshiken to make this a general type.

File: src/components/badge/Badge.tsx [hunk 2]
```diff
@@ -1,40 +1,27 @@
...
-    resizeInput();
-  }, [props.value]);
+export const Badge: React.FC<IBadgeProps> = ({ children, ...props }) => {
 
   return (
     <SBadge>
-      <SBadgeIcon disabled={props.disabled} position={props.position} height={height}>
-        <div ref={badgeRef} style={{ width: "fit-content", visibility: "hidden", position: "absolute", fontSize: 9 }} />
-        <SBadgeIconContent>{props.value}</SBadgeIconContent>
+      <SBadgeIcon {...props}>
```
Comment (diegomart2000) [inline src/components/badge/Badge.tsx]: only pass down required props.

File: src/components/badge/Badge.tsx [hunk 3]
```diff
@@ -1,40 +1,27 @@
...
-      let width = badgeRef.current.scrollWidth;
-
-      setHeight(width + 1);
-    }
-  };
-
-  React.useEffect(() => {
-    resizeInput();
-  }, [props.value]);
+export const Badge: React.FC<IBadgeProps> = ({ children, ...props }) => {
```
Comment (diegomart2000) [inline src/components/badge/Badge.tsx]: Prefer destructuring and React named imports.

**Review by diegomart2000 — CHANGES_REQUESTED** [review]

**Review by diegomart2000 — CHANGES_REQUESTED** [review]
@steiningermona Badges in different styles don't match specifications.

Screen Shot 2021-12-02 at 2 28 26 PM
^^No white borer in specs.

Screen Shot 2021-12-02 at 2 27 25 PM
^^Wrong color

**Comment by steiningermona** [comment]
**from figma**
image

**from storybook**
image
image
image
image
image
image
image

**Review by diegomart2000 — APPROVED** [review]

**Pull request merged**